src/Backend/FakeNews/FakeNews.Bll/Users/UserService.cs

```diff
@@ -1,44 +1,93 @@
 
 namespace FakeNews.Bll.Users
 {
     public class UserService : IUserService
     {
-        private readonly SignInManager<User> signInManager;
         private readonly FakeNewsDbContext dbContext;
         private readonly UserManager<User> userManager;
+        private readonly RoleManager<IdentityRole> roleManager;
+        private readonly IConfiguration configuration;
 
-        public UserService(SignInManager<User> signInManager, FakeNewsDbContext dbContext, UserManager<User> userManager)
+        public UserService(
+            FakeNewsDbContext dbContext,
+            UserManager<User> userManager,
+            RoleManager<IdentityRole> roleManager,
+            IConfiguration configuration)
         {
-            this.signInManager = signInManager;
             this.dbContext = dbContext;
             this.userManager = userManager;
+            this.roleManager = roleManager;
+            this.configuration = configuration;
         }
 
-        public async Task LogUserInAsync(LoginDto loginDto)
+        public async Task<TokenDto> LogUserInAsync(LoginDto loginDto)
         {
             var user = await dbContext.Users.SingleAsync(u => u.Email == loginDto.Email);
-            await signInManager.PasswordSignInAsync(user, loginDto.Password, loginDto.Remember, false);
+
+            var userRoles = await userManager.GetRolesAsync(user);
+            var authClaims = new List<Claim>
+                {
+                    new Claim(ClaimTypes.Name, user.UserName),
+                    new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()),
+                };
+
+            foreach (var userRole in userRoles)
+            {
+                authClaims.Add(new Claim(ClaimTypes.Role, userRole));
+            }
+
+            var authSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(configuration["JWT:Secret"]));
+            var token = new JwtSecurityToken(
+                    issuer: configuration["JWT:ValidIssuer"],
+                    audience: configuration["JWT:ValidAudience"],
+                    expires: DateTime.Now.AddHours(3),
+                    claims: authClaims,
+                    signingCredentials: new SigningCredentials(authSigningKey, SecurityAlgorithms.HmacSha256)
+                    );
+
+            return new TokenDto
+            {
+                Token = new JwtSecurityTokenHandler().WriteToken(token),
+                Expiration = token.ValidTo
+            };
         }
 
         public async Task RegisterUserAsync(RegisterUserDto registerUserDto)
         {
             var user = new User
             {
                 UserName = registerUserDto.Username,
-                Email = registerUserDto.Email
+                Email = registerUserDto.Email,
+                SecurityStamp = Guid.NewGuid().ToString()
             };
 
             await userManager.CreateAsync(user, registerUserDto.Password);
+
+            if (!await roleManager.RoleExistsAsync(UserRoles.Admin))
```

Comment:
This snippet has no place at registration. The database should be seeded instead, and the roles should be entered there as initial values.

Reply:
Moved to seed